webapp/src/components/tutorialValidators.tsx

```diff
@@ -53,21 +61,43 @@ export class BlocksExistValidator extends CodeValidatorBase {
             return defaultResult();
         }
 
+        // if no valid markers are specified, default to "highlight"
+        if (!this.markers || (!this.markers.includes("highlight") && !this.markers.includes("exists"))) {
+            this.markers = ["highlight"];
+        }
+
         const allHighlightedBlocks = await pxt.tutorial.getTutorialHighlightedBlocks(tutorialOptions);
-        if (!allHighlightedBlocks) {
+        const allValidateBlocks = await pxt.tutorial.getTutorialValidateBlocks(tutorialOptions);
+        if (!allHighlightedBlocks && !allValidateBlocks) {
             return defaultResult();
         }
 
         const stepHash = pxt.tutorial.getTutorialStepHash(tutorialOptions);
-        const stepHighlights = allHighlightedBlocks[stepHash];
+        let highlightBlocksCount: pxt.Map<number> = {};
+        let validateBlocksCount: pxt.Map<number> = {};
+
+        if (allHighlightedBlocks && this.markers.includes("highlight")) {
+            highlightBlocksCount = allHighlightedBlocks[stepHash];
+        }
+
+        if (allValidateBlocks && this.markers.includes("exists")) {
```

Comment:
if we refactor out the separate `highlightBlocks` field and just stored in one, this portion would get a little simpler / just a loop over `this.markers` and summing up